sql/catalyst/src/main/java/org/apache/spark/sql/connector/write/BatchWrite.java

```diff
@@ -104,4 +105,19 @@ default void onDataWriterCommit(WriterCommitMessage message) {}
    * clean up the data left by data writers.
    */
   void abort(WriterCommitMessage[] messages);
+
+  /**
+   * Whether this batch write requests merge execution metrics.
+   */
+  default boolean requestMergeMetrics() {
```

Comment:
Is there a performance penalty for requesting metrics? If not, I'd drop this method and always call `commitMerge`.